### ISSUE TITLE
[feature] Uncaught exception should provide a way to get app state

### ISSUE THREAD

**Issue by devkev** [tldraw/tldraw#532]
I was using tldraw.com in Safari on an ipad when I ran into #531.  Unfortunately this means I now can't get access to my drawings, because Safari on ipados doesn't have native devtools (it requires connecting to Safari on macos desktop, which I don't have) - even though I know they're sitting right there in the IndexedDB in the browser.

It would be much better if this generic "a client-side exception has occurred" message also included a way for me to get a copy of the app state. eg. an anchor which when clicked reveals a textarea with the contents of the app state, so I can select-all then copy-paste.

This would help (a) with diagnosing bugs on platforms where there are no native devtools (or they're cumbersome) (ie. mobile), and (b) would also provide users with an "escape hatch" so they can recover their drawings and move them to another system (after correcting any corrupted app state), without needing to go poking around in devtools.

**Comment by steveruizok** [comment]
Are you able to open the app at all? I think you may need to reset the local indexeddb. Totally agree that we should handle bad documents better!

**Comment by devkev** [comment]
I can open the app, but the UI almost instantly is replaced by the error message.  I can indeed reset the storage for tldraw.com (which I assume will include the IndexedDB data), and I expect that will allow me to start fresh again on this device.  But first I wanted to get a copy of the data, so that I could at least get the uncorrupted pages.  I've eventually managed to do that, but it was inordinately difficult.  Even after getting access to a Mac so I could use DevTools on the iPad, it turns out that exporting the IndexedDB requires [jumping through hoops](https://dfahlander.medium.com/export-indexeddb-from-a-web-app-using-devtools-62c55a8996a1).  And in fact it was worse than that, because `URL.createObjectURL` didn't seem to work correctly, async doesn't work in Safari DevTools, DevTools frequently truncates/elides things when you try to copy/save/log them, etc etc.  Eventually I was able to dump the JSON inside `<pre>` and then copy that element out of DevTools and into a text editor (removing the tags).

So it would definitely be a lot better if this error message also had a `<textarea>` with the state document's JSON, for easy copy-pasting.  (Or, you know, at least attempt to - I understand that since an exception has occurred, things could be messed up so badly that it might not be possible to do that.)
